src/Lykke.Service.PaySettlement.Services/AssetService.cs

```diff
@@ -0,0 +1,114 @@
+
+            Asset[] assets = assetGetAllTask.Result.ToArray();
+            AssetPair[] assetPairs = assetPairGetAllTask.Result.Where(p =>
+                    string.Equals(p.BaseAssetId, _settings.PaymentAssetId, StringComparison.OrdinalIgnoreCase)
+                    || string.Equals(p.QuotingAssetId, _settings.PaymentAssetId, StringComparison.OrdinalIgnoreCase))
+                .ToArray();
+
+            foreach (string settlementAssetId in _settings.SettlementAssetIds)
+            {
+                FillAsset(assets, settlementAssetId);
+                FillAssetPair(assetPairs, settlementAssetId);
+            }
+            FillAsset(assets, _settings.PaymentAssetId);
+        }
+
+        private void FillAsset(Asset[] assets, string assetId)
+        {
+            Asset asset = assets.FirstOrDefault(a =>
+                string.Equals(a.Id, assetId, StringComparison.OrdinalIgnoreCase));
+
+            if (asset == null)
+            {
+                _log.Critical(null, $"Asset {assetId} is not found.");
+            }
+
+            _assets[assetId.ToUpper()] = asset;
+        }
+
+        private void FillAssetPair(AssetPair[] assetPairs, string settlementAssetId)
+        {
+            AssetPair assetPair = assetPairs.FirstOrDefault(p =>
+                string.Equals(p.BaseAssetId, settlementAssetId, StringComparison.OrdinalIgnoreCase)
+                || string.Equals(p.QuotingAssetId, settlementAssetId, StringComparison.OrdinalIgnoreCase));
+
+            if (assetPair == null)
+            {
+                _log.Critical(null, $"AssetPair for assets {settlementAssetId} " +
+                                    $"and {_settings.PaymentAssetId} is not found.");
+            }
+
+            _assetPairs[settlementAssetId.ToUpper()] = assetPair;
+        }
+
+        public AssetPair GetAssetPair(string assetPairId)
+        {
+            return _assetPairs.Values.First(p =>
+                string.Equals(p.Id, assetPairId, StringComparison.OrdinalIgnoreCase));
+        }
+
+        public AssetPair GetAssetPair(string paymentAssetId, string settlementAssetId)
+        {
+            if(!string.Equals(paymentAssetId, _settings.PaymentAssetId, StringComparison.OrdinalIgnoreCase ))
+            {
+                throw new ArgumentException($"{nameof(paymentAssetId)} ({paymentAssetId}) " +
+                                            $"does not equal to required {_settings.PaymentAssetId}.");
+            }
+
+            if (!_assetPairs.TryGetValue(settlementAssetId.ToUpper(), out AssetPair assetPair))
+            {
+                throw new ArgumentException($"{nameof(settlementAssetId)} ({settlementAssetId}) " +
+                                            $"is not found in required {_settings.SettlementAssetIds.ToJson()}.");
+            }
+
+            return assetPair;
+        }
+
+        public Asset GetAsset(string assetId)
+        {
+            if (!_assets.TryGetValue(assetId.ToUpper(), out Asset asset))
```

Comment:
This is the same as "return _assets[assetId];"

Reply:
No, I add missed value into exception message.